### DISCUSSION TITLE
shouldUnregister false + disable fields ?

### DISCUSSION THREAD

**Discussion by fadi-george** [react-hook-form/react-hook-form#3221] [Q&A] [answered]
Hello,

I would think that disabled fields shouldn't be submitted when used shouldUnregister false. Seems like a bug?

- click next (twice)
- data submitted includes disabled fields
https://codesandbox.io/s/react-hook-form-695-disabled-fields-fw5ck?file=/src/index.js

**Comment by bluebill1049** [comment, marked as answer]
it's going to be too tricky for this, `shouldUnregister: false` is pretty much like Controlled input, where we keep an object of the entire form value. i would recommend you do the disabled at your app level.

**Reply by fadi-george** [reply]
I see. Can you clarify what you mean, in my actual app I have controls that dynamically disable other fields.

**Reply by bluebill1049** [reply]
`shouldUnregister: true`: we read what's in the form, and if the input is disabled so we can easily omit them.
`shouldUnregister: false`: It's like Controlled input which has a local state of the entire form, so whatever it's initialized there will be in the final submit.

so you may need to do something like `setValue('xxx', undefined)` similar to controlled form, when you setState of the input to be undefined

**Reply by fadi-george** [reply]
I see, thanks will try it out.
